docs/kb/semgrep-ci/azure-using-templates-with-semgrep.md

```diff
@@ -0,0 +1,75 @@
+---
+tags:
+  - Azure
+  - Templates
+description: Using templates in Azure pipelines with Semgrep commands.
+---
+# Using templates in Azure pipelines with Semgrep
+
+## Motivation
+
+Complex code can take more work to maintain and modify. So, encapsulating some functionalities in functions, methods, or external files is a good programming practice.
+That is the primary goal of Azure templates: extract some functionalities (snippets) defined in a big yaml file to an external file (template). With that, we are improving pipelines' readability and maintainability.
```

Comment:
```suggestion
The primary goal of Azure templates is to extract some functionalities defined in a big yaml file and place these into an external file (the [template](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops)). This improves a pipeline's readability and maintainability.
```

I removed "snippets" because the official documentation also refers to them as "reusable content" and code snippets are a generic term while these templates are specific to Azure Pipelines. https://learn.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops